rules/prefer-ternary.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+const getDocumentationUrl = require('./utils/get-documentation-url');
+const isParenthesized = require('./utils/is-parenthesized');
+
+const create = context => {
+	function checkIfStatement(node) {
+		if (isSingleBlockStatement(node) &&
+		isSameType(node) &&
+			checkConsequentAndAlternateType(node)) {
+			context.report({
+				node,
+				message: 'This `if` statement can be replaced by a ternary operator.',
+				fix(fixer) {
+					return fixFunction(node, fixer);
+				}
+			});
+		}
+	}
+
+	function fixFunction(node, fixer) {
+		const sourceCode = context.getSourceCode();
+		let prefix = '';
+		const ifCondition = sourceCode.getText(node.test);
+		let left = '';
+		let right = '';
+		if (getNodeBody(node.consequent).type === 'ExpressionStatement') {
+			const expressionType = getNodeBody(node.consequent).expression.type;
+			if (expressionType === 'AssignmentExpression') {
+				prefix = sourceCode.getText(getNodeBody(node.consequent).expression.left) + ' = ';
+				left = getNodeBody(node.consequent).expression.right;
+				right = getNodeBody(node.alternate).expression.right;
+			} else {
+				if (expressionType === 'AwaitExpression') {
+					prefix = 'await ';
+				} else {
+					prefix = getNodeBody(node.consequent).expression.delegate ? 'yield* ' : 'yield ';
+				}
```

Comment:
I don't think logic is not right.

We need check `YieldExpression`.

Why it has to be `await` and `yield`? There is other cases.

Reply:
Since this is in the fixFunction, it would not reach that line unless it was await or yield